backend/tests/test_offline_guard.py

```
@@ -0,0 +1,109 @@
+
+def _tf_hub():
+    import transformers.utils.hub as tf_hub
+
+    return tf_hub
+
+
+def test_mutates_cached_huggingface_hub_constant():
+    original = _hf_const().HF_HUB_OFFLINE
+    with force_offline_if_cached(True, "t"):
+        assert _hf_const().HF_HUB_OFFLINE is True
+    assert _hf_const().HF_HUB_OFFLINE == original
```

Ruff (0.15.10)

[warning] Yoda condition detected

Rewrite as `original == _hf_const().HF_HUB_OFFLINE`

(SIM300)